src/app/card-search.tsx

```diff
@@ -22,7 +22,7 @@ interface CardSearchProps {
 }
 
 export const CardSearch: FC<CardSearchProps> = ({ initialQuery = '' }) => {
-  const [searchResults, setSearchResults] = useState<Card[]>([]);
+  const [searchResults, setSearchResults] = useState<Card[][]>([]);
```

Comment:
I'm not sure I'm a fan of how I made the nested indexing implicit here. I wonder if it'll be better if I renamed the variable to something more explicit like `searchResultsByPage` if we think this is the right overall approach.

Reply:
I'm not convinced it's the "right overall approach" yet, simply because I haven't tested it myself.

However, I have some thoughts on naming/variables:
- Prefer `pagedSearchResults` over `searchResultsByPage`, because the latter implies sorting that isn't relevant here
- Prefer explicit typing over implicit typing. Instead of using the implicit type `Card[][]`, I would prefer we define an explicit type like below:
```ts
type PagedSearchResults = Record<number, Card[]>;
```

I understand that defining that type would represent a change from an array-of-array to a dict-of-array, but this is preferred as the latter aligns better with my interpretation of Typescript conventions.

Combining the above points, we end up with the following:
```suggestion
  const [pagedSearchResults, setPagedSearchResults] = useState<PagedSearchResults>({});
```